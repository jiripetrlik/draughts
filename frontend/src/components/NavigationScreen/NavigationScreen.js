import React from 'react';
import Accordion from '@material-ui/core/Accordion';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { sendMessage } from '../../api/index'

class NavigationScreen extends React.Component {
    waitForOponent() {
        let command = {
            Name: "wait",
            Parameters: ""
        }

        sendMessage(JSON.stringify(command))
    }

    render() {
        let waitingPlayers = this.props.players.Waiting.map((player) =>
            <ListItem button key={player.Id}>{player.Name}</ListItem>
        )
        let playingPlayers = this.props.players.Playing.map((players) =>
            <ListItem button key={players.id}>{players.names[0]} vs {players.names[1]}</ListItem>
        )

        return (
            <div className="navigation-screen">
                <Button onClick={this.waitForOponent}>Wait for oponent</Button>
                <Accordion defaultExpanded={true}>
                    <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                        <Typography>Waiting for oponent</Typography>
                    </AccordionSummary>
                    <AccordionDetails>
                        <List>
                            {waitingPlayers}
                        </List>
                    </AccordionDetails>
                </Accordion>

                <Accordion defaultExpanded={true}>
                    <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                        <Typography >Now playing</Typography>
                    </AccordionSummary>
                    <AccordionDetails>
                        <List>
                            {playingPlayers}
                        </List>
                    </AccordionDetails>
                </Accordion>
            </div>
        )
    }
}

export default NavigationScreen
