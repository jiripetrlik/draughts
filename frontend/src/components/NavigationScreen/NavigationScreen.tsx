import React from 'react';
import Accordion from '@material-ui/core/Accordion';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { sendMessage } from '../../api/index'

class NavigationScreen extends React.Component<Readonly<any>, Readonly<any>> {

    constructor(props: any) {
        super(props)
        this.waitForOponent = this.waitForOponent.bind(this)
    }

    waitForOponent() {
        let command = {
            Name: "wait",
            Parameters: ""
        }

        sendMessage(JSON.stringify(command))
    }

    joinOponent(oponentID: number) {
        let command = {
            Name: "join",
            Parameters: oponentID.toString()
        }

        sendMessage(JSON.stringify(command))
    }

    render() {
        let waitingPlayers = this.props.players.Waiting.map((player: any) =>
            <ListItem button key={player.ID} onClick={this.joinOponent.bind(this, player.ID)}>{player.Name}</ListItem>
        )
        let playingPlayers = this.props.players.Playing.map((players: any) =>
            <ListItem button key={players.ID}>{players.Names[0]} vs {players.Names[1]}</ListItem>
        )

        return (
            <div className="navigation-screen">
                <Button onClick={this.waitForOponent}>Wait for oponent</Button>
                <Accordion defaultExpanded={true}>
                    <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                        <Typography>Waiting for oponent</Typography>
                    </AccordionSummary>
                    <AccordionDetails>
                        <List>
                            {waitingPlayers}
                        </List>
                    </AccordionDetails>
                </Accordion>

                <Accordion defaultExpanded={true}>
                    <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                        <Typography >Now playing</Typography>
                    </AccordionSummary>
                    <AccordionDetails>
                        <List>
                            {playingPlayers}
                        </List>
                    </AccordionDetails>
                </Accordion>
            </div>
        )
    }
}

export default NavigationScreen
