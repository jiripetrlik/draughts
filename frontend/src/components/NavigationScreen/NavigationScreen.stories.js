import React from 'react'
import NavigationScreen from './NavigationScreen'

export default {
  component: NavigationScreen,
  title: 'NavigationScreen',
};

const Template = args => <NavigationScreen {...args} />;

export const Default = Template.bind({});
Default.args = {
  players: {
    Waiting: [

    ],
    Playing: [

    ]
  },
};

export const Filled = Template.bind({});
Filled.args = {
  players: {
    Waiting: [
      {
        ID: 1,
        Name: "User 1"
      },
      {
        ID: 2,
        Name: "User 2"
      }
    ],
    Playing: [
      {
        ID: 1,
        Names: ["User 3", "User 4"]
      },
      {
        ID: 2,
        Names: ["User 5", "User 6"]
      },
      {
        ID: 3,
        Names: ["User 7", "User 8"]
      },
    ]
  },
};
