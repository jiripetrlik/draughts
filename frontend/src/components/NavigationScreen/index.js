import NavigationScreen from "./NavigationScreen"

export default NavigationScreen
