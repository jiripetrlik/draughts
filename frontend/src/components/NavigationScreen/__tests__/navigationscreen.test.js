import React from 'react';
import renderer from 'react-test-renderer';
import NavigationScreen from '../NavigationScreen'

test('snapshot test', () => {
    const players = {
        Waiting: [
            {
                Id: 1,
                Name: "User 1"
            },
            {
                Id: 2,
                Name: "User 2"
            }
        ],
        Playing: [
            {
                id: 1,
                names: ["User 3", "User 4"]
            },
            {
                id: 2,
                names: ["User 5", "User 6"]
            },
            {
                id: 3,
                names: ["User 7", "User 8"]
            },
        ]
    }

    const tree = renderer
        .create(<NavigationScreen players={players} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
