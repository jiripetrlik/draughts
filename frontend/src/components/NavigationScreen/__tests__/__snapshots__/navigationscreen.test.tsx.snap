// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="navigation-screen"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Wait for oponent
    </span>
  </button>
  <div
    className="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content Mui-expanded"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Waiting for oponent
        </p>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root"
            >
              <ul
                className="MuiList-root MuiList-padding"
              >
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  User 1
                </div>
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  User 2
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content Mui-expanded"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Now playing
        </p>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root"
            >
              <ul
                className="MuiList-root MuiList-padding"
              >
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  User 3
                   vs 
                  User 4
                </div>
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  User 5
                   vs 
                  User 6
                </div>
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  User 7
                   vs 
                  User 8
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
